src/main/java/com/sap/sgs/phosphor/fosstars/tool/github/AbstractReporter.java

```diff
@@ -1,7 +1,80 @@
 package com.sap.sgs.phosphor.fosstars.tool.github;
 
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.sap.sgs.phosphor.fosstars.tool.Project;
 import com.sap.sgs.phosphor.fosstars.tool.Reporter;
+import java.io.IOException;
+import java.net.URL;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
 
-abstract class AbstractReporter implements Reporter {
+abstract class AbstractReporter<T extends Project> implements Reporter<T> {
+
+  /**
+   * For serialization.
+   */
+  private static final ObjectMapper MAPPER = new ObjectMapper();
+
+  /**
+   * A type reference of a list of {@link GitHubProject}s for deserialization.
+   */
+  private static final TypeReference<List<GitHubProject>> LIST_OF_GITHUB_PROJECTS_TYPE
+      = new TypeReference<List<GitHubProject>>() {};
+
+  /**
+   * Merges two lists of projects. The merge is based on URLs.
+   * If there are two projects with the same URLs,
+   * then the method uses the project from the first list.
+   *
+   * @param projects The first list of projects.
+   * @param extraProjects The second list of projects.
+   * @return A list of projects which contains all projects from the original lists.
+   */
+  static List<GitHubProject> merge(
```

Comment:
Doesn't GitHubProject have a equals method? with that, we can simply handle this with 
`
for (Object x : two){
   if (!one.contains(x))
      one.add(x);
}
`
?

Reply:
Yes, `GitHubProject` has `equals()` and `hashCode()` methods. Besides project and organization names, the class  also contains a rating value. The rating value should not be taken into account in the `merge()` method. However, the current implementations of `equals()` and `hashCode()` do not use the rating value. So, your suggestion would work. If `equals()` and `hashCode()` method change in the future, the `merge()` method may stop working as expected. I would like to avoid that, so that I implemented it with an extra map. Please let me know if you see anything wrong here.

Reply:
Makes sense @artem-smotrakov thanks for the information.